corehq/apps/domain/management/commands/unlink_project_spaces.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from django.core.management.base import BaseCommand
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.linked_domain.models import DomainLink
+from corehq.apps.app_manager.models import Application
+from six.moves import input
+
+
+class Command(BaseCommand):
+    help = "Unlinks linked project spaces and converts the downstream app into a standalone app."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'master_domain_name',
+            help='The name of the master project space'
+        )
+        # parser.add_argument(
+        #     'master_app_name',
+        #     help='The name of the master app'
+        # )
+        parser.add_argument(
+            'linked_domain_name',
+            help='The name of the linked project space'
+        )
+        parser.add_argument(
+            'linked_app_name',
+            help='the name of the linked app'
+        )
+
+    # def handle(self, master_domain_name, master_app_name, linked_domain_name, linked_app_name, **options):
+    def handle(self, master_domain_name, linked_domain_name, linked_app_name, **options):
+        master_domain = Domain.get_by_name(master_domain_name)
+        linked_domain = Domain.get_by_name(linked_domain_name)
+        if not master_domain:
+            print('domain with name "{}" not found'.format(master_domain_name))
+            return
+        if not linked_domain:
+            print('domain with name "{}" not found'.format(linked_domain_name))
```

Comment:
I'm not sure these checks are necessary, as you really only care that a `DomainLink` with those domain names exist. I can imagine a scenario where a domain is deleted, but there still exists a `DomainLink` that we'd want to delete.

Reply:
Is there a way to transfer an app to a different domain? Or can an app exist without its original domain?

I don't think that we should be un-linking apps that are now deleted or when their domain has been deleted (assuming deleting a domain deletes its apps). And this command doesn't really delete/remove the `DomainLink`; it marks it as `deleted` so these apps don't show up as linked in the project spaces's Linked Projects page.

Reply:
Oh wait, do you mean that there may be a situation in which someone wants to convert break the link for a downstream app, when the master app has been deleted?

Reply:
yes, that's exactly what I was imagining. sorry for being cryptic!